src/huggingface_hub/keras_mixin.py

```diff
@@ -26,10 +27,27 @@
     import tensorflow as tf
 
 
+def flatten_dict(dictionary, parent_key="", sep="."):
```

Comment:
Parameter descriptions would be nice here. Also, if we aren't using `parent_key` or `sep`, should we drop those params?

Reply:
also is there a risk here that something within this dict might not be serializable? Just curious.

Reply:
> Parameter descriptions would be nice here. Also, if we aren't using parent_key or sep, should we drop those params?

I think if we decide on using `.` as the seperator, we can drop the `sep` parameter. We cannot drop the `parent_key` as this forms the basis of recursing over the parent keys and joining it with child keys in the nested dictionary.

Reply:
> also is there a risk here that something within this dict might not be serializable? Just curious.

IMO we should not have anything un-serializable as this was packed inside the `model.optimizer` as the configuration. But I might be wrong here as well.